### PR TITLE
Fix where access log is not written when the method of a reques…

### PR DESCRIPTION
**Pull request by minwoox**
…ot allowed

Motivation:
There's some cases that we don't call `RequestLogBuilder.endRequest()`.
- When the method of the request is not allowed
- When there are no services that match the path of the request
- When the path of the request is invalid
- and so on

Because the `RequestLog` is never complete, `AccessLogWriter` is not called so we should fix it.

Modification:
- Call `RequestLogBuilder.endRequest()` even when `service.serve()` is not invoked

Result:
- You can now see the access logs when the method of a request is not allowed and there are no services that match the path.

### CODE REVIEW

File: core/src/test/java/com/linecorp/armeria/server/HttpServerHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    void handleNonExistentMapping() {
+        final HttpClient client = HttpClient.of(server.uri("/"));
+        final AggregatedHttpResponse res = client.get("/non_existent").aggregate().join();
+        assertThat(res.status()).isSameAs(HttpStatus.NOT_FOUND);
+        await().untilAsserted(() -> assertThat(logHolder.get().path()).isEqualTo("/non_existent"));
+        assertThat(logHolder.get().requestCause()).isNull();
+    }
+
+    @Test
+    void httpStatusException() {
```
Comment (minwoox) [inline core/src/test/java/com/linecorp/armeria/server/HttpServerHandlerTest.java]: I found out that we leave the `httpStatusException` as the cause but not for `httpResponseException`.
I think we should treat them in the same way and I think we do not log the causes in both cases.

Comment (trustin) [reply core/src/test/java/com/linecorp/armeria/server/HttpServerHandlerTest.java]: Agreed. Could you update accordingly?

Comment (minwoox) [reply core/src/test/java/com/linecorp/armeria/server/HttpServerHandlerTest.java]: Fixed. PTAL 🙇

**Comment by trustin** [comment]
@ikhoon Still LGTY?

**Comment by minwoox** [comment]
Thanks for reviewing! 😉

**Pull request merged**